post-processor/vagrant-cloud/client_test.go

```diff
@@ -0,0 +1,29 @@
+package vagrantcloud
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+)
+
+func TestVagranCloudErrors(t *testing.T) {
+	testCases := []struct {
+		resp     string
+		expected string
+	}{
+		{`{"Status":"422 Unprocessable Entity", "StatusCode":422, "errors":[]}`, ""},
+		{`{"Status":"404 Artifact not found", "StatusCode":404, "errors":["error1", "error2"]}`, "error1. error2"},
+		{`{"StatusCode":403, "errors":[{"message":"Bad credentials"}]}`, "message Bad credentials"},
```

Comment:
I reached out to the Vagrant team about this error format. It is possible that it will get converted to just an array of strings, which will greatly simplify the logic for parsing out errors.